### DISCUSSION TITLE
Kibana errors : "the field associated with this object no longer exists in the index pattern.

### DISCUSSION THREAD

**Discussion by Mastadamus** [Security-Onion-Solutions/securityonion#4714] [Unsupported Versions]
I have a CentOs based build from the official ISO. I recently upgraded to 2.3.60 hotfix via Soup. When I go into Kibana, I noticed that a good deal of data isn't showing and i get errors instead such as "the field process.command_line.keyword associated with this object no longer exists in the index pattern. I get several similar errors for the "file" section of kibana and all of the strelka stuff is errored out.  Does anyone know why or how I can fix this?
![image](https://user-images.githubusercontent.com/79027361/124660179-7920ed00-de6b-11eb-9aff-c0f6f4a28b74.png)

Security Onion 2.3.60 with hot fix 
CentOS
Ryzen threadripper 
16gb ram. 
1tb ssd.

**Comment by golfreeze** [comment]
please try on

# ls -la /opt/so/conf/logstash/pipelines/manager
total 20
drwxr-xr-x 2 logstash socore 4096 May 22 23:57 .
drwxr-xr-x 4 logstash socore 4096 May 29 23:42 ..
-rw-r--r-- 1 logstash socore   69 May 22 23:57 0009_input_beats.conf
-rw-r--r-- 1 logstash socore 1065 May 22 23:57 0010_input_hhbeats.conf
-rw-r--r-- 1 logstash socore  206 May 22 23:57 9999_output_redis.conf

# ls -la /opt/so/conf/logstash/pipelines/search
total 44
drwxr-xr-x 2 logstash socore 4096 May 29 23:42 .
drwxr-xr-x 4 logstash socore 4096 May 29 23:42 ..
-rw-r--r-- 1 logstash socore  180 May 29 23:42 0900_input_redis.conf
-rw-r--r-- 1 logstash socore  372 May 29 23:42 9000_output_zeek.conf
-rw-r--r-- 1 logstash socore  351 May 29 23:42 9002_output_import.conf
-rw-r--r-- 1 logstash socore  342 May 29 23:42 9034_output_syslog.conf
-rw-r--r-- 1 logstash socore  392 May 29 23:42 9100_output_osquery.conf
-rw-r--r-- 1 logstash socore  341 May 29 23:42 9400_output_suricata.conf
-rw-r--r-- 1 logstash socore  370 May 29 23:42 9500_output_beats.conf
-rw-r--r-- 1 logstash socore  338 May 29 23:42 9600_output_ossec.conf
-rw-r--r-- 1 logstash socore  359 May 29 23:42 9700_output_strelka.conf

2. add and check content on file /opt/so/saltstack/local/pillar/global.sls
logstash:
  pipelines:
    manager:
      config:
        - so/0009_input_beats.conf
        - so/0010_input_hhbeats.conf
        - so/9999_output_redis.conf.jinja
    search:
      config:
        - so/0900_input_redis.conf.jinja
        - so/9000_output_zeek.conf.jinja
        - so/9002_output_import.conf.jinja
        - so/9034_output_syslog.conf.jinja
        - so/9100_output_osquery.conf.jinja
        - so/9400_output_suricata.conf.jinja
        - so/9500_output_beats.conf.jinja
        - so/9600_output_ossec.conf.jinja
        - so/9700_output_strelka.conf.jinja
3. restart logstash
# so-logstash-restart

4. restart elasticsearch
# so-elasticsearch-restart

**Reply by Mastadamus** [reply]
None of the logtash code is present in my global.sls but I'm unsure of the proper indent/spacing to add what you posted. when I just copy and paste it it won't allow me to restart logtash it errors out because of this.  Can you include a screenshot of what the formatting should look like or post example with proper indentation if thats my issue?

**Reply by Mastadamus** [reply]
Nevermind, I found the code example in the Documentation. I'm trying it now.

**Comment by Mastadamus** [comment]
After adding that logtash config to the Global.sls, I was able to restart elastic and logtash. Unfortunately, That didn't solve my problem. It seems no new data is making it into elastic/kibana. I show no logs ingested in kibana and have zero alerts since the upgrade to 2.3.60

**Reply by weslambert** [reply]
Are there any errors/clues if you tail the Logstash log in `/opt/so/log/logstash/logstash.log`?

**Reply by Mastadamus** [reply]
Sorry, I rolled back to previous image before I saw this. When I attempt to upgrade again, I will check this log IF i run into the problem again.

**Comment by Nicalicious** [comment]
Seems a duplicate thread. Posting in both.

Any update on this? I'm seeing this exact error on several of the dashboards within Kibana. RDP/SMB/Intel/IRC/Modbus/MySQL/NTLM/PE/RADIUS/RDP/and so on...

-Fresh install from latest .iso 2.3.61
-ran sudo soup
-replaced certificate with self-signed certificate /etc/pki/managerssl.crt and /etc/pki/managerssl.key

Everything else is default installation for Standalone.

-vcenter/esxi build
-2 NICs /one management / one for promiscuous mode
-32GB RAM
-24 cpu xeon X5670
-SSD storage

**Reply by weslambert** [reply]
The dashboard errors are likely because there is no data (with that field) within Elasticsearch to populate the mappings that Kibana uses to tie to its visualizations.

**Reply by Nicalicious** [reply]
This doesn't seem to be the case. It seems the mappings should be pre-populated in the installation and are missing. Trying to manually add them to the index patter proves fruitless.

**Reply by Nicalicious** [reply]
I have tried rebuilding the installation many times. It is the 2.3.61 iso that is available. Same effect every time.

**Comment by fluffyunicorn24** [comment]
I recently had this issue and I am not sure if this helps but here is what I did:

In the Elastic Dashboard I went to Stack Management > Kibana > Index Patterns
Locate the default index pattern and select Add Field and put the field it is looking for.   

Again, I am not sure if this is the right solution, but it was the one for me.

**Reply by Nicalicious** [reply]
![radius add](https://user-images.githubusercontent.com/38387334/128275796-065df08d-e1d6-4fb7-9594-0afb8cffbb5f.png)
I tried that as well. It removes the error but does not correct the issue and shows no data for the dashboard/card. I added the keyword for RADIUS as it is easier to replicate.

**Reply by weslambert** [reply]
The dashboard errors are likely because there is no data (with that field) within Elasticsearch to populate the mappings that Kibana uses to tie to its visualizations.

**Reply by Nicalicious** [reply]
I thought that as well. I used RADIUS as it was easy to recreate the traffic on-demand. There should be data for the field. It would have to be able to see the reply if it can see the "to" and "from" data as well as the username used. It just seems the keyword search that it is looking for is missing from the index pattern "*:so-*" (default). 
![radius with data](https://user-images.githubusercontent.com/38387334/128716536-5cdd73ec-0301-4ca7-b56a-ee2d760f6518.png)

This issue is happening with dozens of fields.

**Reply by corycandia** [reply]
Same here.  Though I am fairly novice at SO 2.3, it doesn't appear to just be limited to some types of data.

Maybe I'm just doing this wrong and the older sections are no longer used?  Here's my screen shots attached.
![BRO - Syslog](https://user-images.githubusercontent.com/88662234/128734570-bb6f1811-aa66-43c1-86c1-d50ecf42dc00.PNG)
![SO-Good](https://user-images.githubusercontent.com/88662234/128734577-f87bc1fc-8b88-459f-8c04-99616ea3b864.PNG)

![stack-management_SO](https://user-images.githubusercontent.com/88662234/128734581-6bae5499-de8b-40af-8bb6-bb553356943e.PNG)

![stack-management_LS](https://user-images.githubusercontent.com/88662234/128734555-c93a9026-742b-4fb5-97e4-1efc825448c7.PNG)
![elastic](https://user-images.githubusercontent.com/88662234/128734557-baef086f-05ba-4282-b1ad-fca482ed3c2e.PNG)
![NIDS](https://user-images.githubusercontent.com/88662234/128734562-14fee8bc-ef82-4ce3-bdd6-ea5768edf2d8.PNG)
![BRO-Con](https://user-images.githubusercontent.com/88662234/128734564-46dcb37f-98df-4018-b2cb-79e2c3737bb0.PNG)
![BRO - DNS](https://user-images.githubusercontent.com/88662234/128734567-06dda5bb-b3e2-4d25-bdba-353ee40dc3d0.PNG)
![Disc-SO-Good](https://user-images.githubusercontent.com/88662234/128734568-c2cfa810-56a9-4f94-a6f7-d15d9d17c55b.PNG)
![LogStash](https://user-images.githubusercontent.com/88662234/128734569-9fe45572-4507-49dd-8108-7938b62ed88b.PNG)

**Reply by weslambert** [reply]
The reason for the errors with the logstash-prefixed patterns and dashboards are probably because there are no `logstash-` -prefixed indices.  Some of those dashboards that you mentioned (ex. Bro) are from older installations, and you would only have that data if you upgraded from 16.04.

**Reply by weslambert** [reply]
However, there may be other cases where the mapping has changed/or is not being applied to certain fields, so it's not aligning with what the visualization is looking for.

**Reply by corycandia** [reply]
> The reason for the errors with the logstash-prefixed patterns and dashboards are probably because there are no `logstash-` -prefixed indices. Some of those dashboards that you mentioned (ex. Bro) are from older installations, and you would only have that data if you upgraded from 16.04.

It would make sense if it was an upgrade, but it was a new deployment, completely blank disks (VMDKs).  Should we just remove them, or are those selections supposed to work?  Maybe those selections are just artifacts from previous versions that will be removed?

From my below post, my environment:
This appears to still be happening during my new installation.

Clean install 2.3.61 / Installed from ISO / CentOS / 8GB RAM
SearchMaster
SO-Status = All good
Grid OK

**Reply by weslambert** [reply]
You can remove them (logstash prefixed index patterns) if you like --it won't hurt anything.  Even on new installs, the same index patterns are pre-populated (even though there will be no new data generated to match it -- this is by design, so that folks who are upgrading will still be able to easily see their old data...I'm sure we will phase this out at some point, if not very soon), which is why you are seeing the message.

**Comment by corycandia** [comment]
This appears to still be happening during my new installation.

Clean install 2.3.61 / Installed from ISO / CentOS / 8GB RAM
SearchMaster
SO-Status = All good
Grid OK

[example logstash entries.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6954239/example.logstash.entries.txt)

Found some entries in the log, looks related to indexing

Any movement on this?

**Reply by weslambert** [reply]
The dashboard errors are likely because there is no data (with that field) within Elasticsearch to populate the mappings that Kibana uses to tie to its visualizations.

**Reply by Nicalicious** [reply]
This just doesn't seem to be the case. These fields seem like they should be pre-populated in the installation. The traffic has definitely been ingested. Is there a way to properly rebuild the so-* index?

**Reply by weslambert** [reply]
We were pre-populating the index pattern with field mappings --  however, the behavior with regard to mappings has changed in more recent versions of the Elastic stack.  Kibana now gets information about mappings from Elasticsearch's Field Caps (Field Capabilities) API. This means that it checks with ES to determine what a field's capabilities/mapping(s) is/are, then uses that information to determine how data should be formatted/presented within Kibana. 

If there is no data within Elasticsearch from which Kibana would derive this information, then there is no way it can know of a mapping/field that was tied to a pre-existing visualization (again, until that data is indexed inside of Elasticsearch). The information for the index pattern ( visible from within Kibana) is also continually refreshed (which would overwrite the pre-populated mappings). The "index"  you are referring to rebuilding is not necessarily an index, but an index pattern that Kibana references to determine the format/type of data.  The actual index/data resides within Elasticsearch.

**Reply by Nicalicious** [reply]
![radius with data](https://user-images.githubusercontent.com/38387334/129418194-96cfd67b-6829-4a80-8cb9-01e874d57810.png)
Ok so with that said. How do we correct the issue? The data is definitely ingested. In the screenshot above, it shows the RADIUS source, destination, username, and the result. It knows the result but errors out with ""radius.reply_message.keyword" associated with this object no longer exists in the index pattern". This is a brand new fresh installation. If the NIC is able to sniff all the data including the result, why would it not populate the field for "Reply"? Just trying to figure out how to get this to work. Several sections in the "Network" dashboard of Kibana simply show the same error. 

Sidenote: I ran a vulnerability scan against my network to populate extra data as well. Still no improvement.

**Reply by weslambert** [reply]
First, you'll want to check to see if that field is even populated in an event.  If you expand the records under `All Logs` do you see the field (radius.reply_message.keyword) in any of the results (or an unparsed/unindexed, differently named field that would be it)?

**Reply by Nicalicious** [reply]
![index pattern radius results](https://user-images.githubusercontent.com/38387334/129419712-e44f3ac3-a9cf-4b93-b275-899a89321a3c.png)
![message keyword](https://user-images.githubusercontent.com/38387334/129419716-1727b9a1-18c0-4613-85ad-bc60edd30d01.png)

It seems that it does not populate in the raw logs. The closest data is "message.keyword" without the "radius.reply" prefix.

This data comes from a Microsoft RADIUS server.

**Reply by weslambert** [reply]
If there is no `reply_msg` in the actual Zeek log, then you are not going to see that visualization become populated.

From the Zeek logs page for RADIUS:

```
reply_msg: string&log&optional
Reply message from the server challenge. This is frequently shown to the user authenticating.
```
https://docs.zeek.org/en/v4.0.0/scripts/base/protocols/radius/main.zeek.html

This is how the field gets renamed in the ingest node pipeline for Elasticsearch:

`{ "rename":         { "field": "message2.reply_msg",        "target_field": "radius.reply_message", "ignore_missing": true  } }`

At this time, it sounds like you are just not getting data/logs that contain that field, so there is no `radius.reply_message` field being indexed into the Elasticsearch index.

**Reply by Nicalicious** [reply]
It seems that you are saying there is no solution. The RADIUS traffic is pretty standard traffic from a Microsoft RADIUS server. RADIUS is just the tip of the iceberg for errors in Kibana/Elasticsearch. I just chose to use RADIUS to troubleshoot as it was easier to replicate the traffic on demand. These errors are on several sections of Network Data.  

So are these fresh installs of 2.3.61 iso are just out of luck? Security Onion 16.04 didn't have these issues. It appears that multiple people are having these same issues. Should a patch/hotfix be expected or would it be better to downgrade back to 16.04?

**Reply by weslambert** [reply]
In a technical sense, it's behaving as intended (you just don't have any data to populate the visualization). The issue you've mentioned is more of an aesthetic nuisance. As I mentioned, the behavior of mappings/how Kibana ascertains the mappings has changed (from the Elastic perspective, not controlled by us). In 16.04, we could apply a predefined set of mappings, and unless you refreshed those, the visualizations would not display the same kind of message.  You can absolutely copy the dashboard and remove the visualization if you like. It may be best for us to just remove the visualization (and other visualizations that might rely on less-common fields) if it causes confusion for everyone.  

I don't think that would completely eliminate seeing the visualizations with the same message, however. It doesn't really change anything about any of your operations within Kibana. If the visualization(s) weren't there, you wouldn't know the difference, because the data itself is not there (and you can't act on it anyway).  This might bubble up to being more of a user education/expectation exercise, or consideration for restructuring the  current state of the dashboards.  Ultimately, the dashboards are intended to be an example from which you can copy and modify to suit your environment.

**Reply by corycandia** [reply]
>Ultimately, the dashboards are intended to be an example from which you can copy and modify to suit your environment.

For a SO novice, this makes a lot more sense now.  If you'll tolerate a thought from a 20+ year infrastructure veteran, but a novice in the SO/linux/open source world...

While I think a large majority of the security and monitoring community are intermediate or advanced IT professionals who either have the skill set to easily customize the SO solution, or at least expect that they'll need to, the novice or beginner professional may not.  

Likewise, if a company is testing out a SO solution/implementation for the first time and comparing it to other solutions in the market, they may have a less than ideal experience if the 'out of the box' installation is still in migration limbo and 'looks' broken, even if it's not.  

Just food for thought.  God bless SO.
